aurora-postgresql-plugin/src/test/java/co/cask/auroradb/postgres/AuroraPostgresPluginTestBase.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.auroradb.postgres;
+
+import co.cask.ConnectionConfig;
+import co.cask.DBRecord;
+import co.cask.cdap.api.artifact.ArtifactSummary;
+import co.cask.cdap.api.plugin.PluginClass;
+import co.cask.cdap.datapipeline.DataPipelineApp;
+import co.cask.cdap.proto.id.ArtifactId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.test.TestConfiguration;
+import co.cask.db.batch.DatabasePluginTestBase;
+import co.cask.db.batch.sink.ETLDBOutputFormat;
+import co.cask.db.batch.source.DataDrivenETLDBInputFormat;
+import com.google.common.base.Charsets;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.postgresql.Driver;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Map;
+import java.util.TimeZone;
+
+public class AuroraPostgresPluginTestBase extends DatabasePluginTestBase {
+  protected static final ArtifactId DATAPIPELINE_ARTIFACT_ID = NamespaceId.DEFAULT.artifact("data-pipeline", "3.2.0");
+  protected static final ArtifactSummary DATAPIPELINE_ARTIFACT = new ArtifactSummary("data-pipeline", "3.2.0");
+  protected static final long CURRENT_TS = System.currentTimeMillis();
```

Comment:
seen this a few times now. Should we move to the base class?

Reply:
I struggled a lot with setting up test framework (due to its static nature) when working on our very first MySQL plugin. There was a workaround with startCount in original database-plugins tests. The DATAPIPELINE_ARTIFACT was registered in base class once for the whole test suite and when other plugin's tests where trying to set up a test pipeline with their own Sources and Sinks the error occurred that DATAPIPELINE_ARTIFACT was already registered. So I had to duplicate this code and register it for every plugin's test group, while leaving only static utility methods in base tests class (I mean the class that all tests inherit from). 

On the other hand, this allows us to run tests for one or several particular modules at a time instead of triggering build for the whole database-plugins project.

Reply:
Hmm ok sg